app.js

```diff
@@ -1,151 +0,0 @@
-document.addEventListener("DOMContentLoaded", () => {
-  let comments = [];
-  // Target DOM Elements
-  let comment_box = document.querySelector("#comment_box");
-  let comment_btn = document.querySelector("#add_comment");
-  let comment_wrapper = document.querySelector("#comment-listing");
-  let reply_textbox = document.querySelector("#reply_comment");
-  let nestedCommentForm = document.querySelector(
-    "#comment_list_inner .comment-area"
-  );
-
-  //initialize loacl stroge if not
-  if (localStorage.getItem("comments") === null)
-    localStorage.setItem("comments", JSON.stringify(comments));
-
-  //Add comment object in localStorge
-  let addComment = (comment) => {
-    let all_comments = getAllComments();
-    all_comments.push(comment);
-    updateStroge(all_comments);
-  };
-
-  //update local stroge with new data
-  let updateStroge = (newData) => {
-    localStorage.setItem("comments", JSON.stringify(newData));
-    renderComments();
-  };
-
-  //Add reply on the same index
-  let addReply = (reply, index) => {
-    let all_comments = getAllComments();
-    all_comments.forEach((element, i) => {
```

Comment:
Instead of combining everything in one forEach call, do it as below
1. Find the comment to be updated using `Array.prototype.find`
2. Update that comment
3. Save that comment.

Reply:
I used here .find() method and it also working fine 

  all_comments.find((element, i) => {
      if (i == index) {
        if (element.children) {
          element.children.push(reply);
        } else {
          element.children = [];
          element.children.push(reply);
        }
      }
    });


what was the issue in ForEach ?

Reply:
Don't do all this inside `.find`. use `find` to just find the object. Read documentation of Array.prototype.find.